webapp/src/ts/services/rules-engine.service.ts

```diff
@@ -154,6 +155,11 @@ export class RulesEngineService implements OnDestroy {
                 initializeTelemetryData.record();
               });
           });
+      })
+      .catch(error => {
+        const errorMessage = `Failed to initialize RulesEngineService`;
+        this.feedbackService.submit(errorMessage);
```

Comment:
The way the feedback service works is that it will include the last 10 entries that have been logged. If you catch the error here, before it gets logged, you will end up with a feedback document that is actually missing the actual error: 
![image](https://user-images.githubusercontent.com/35681649/230384749-9fc41d03-51c0-4713-86ce-7d852ed7d5ef.png)

To test this, I manually added some gibberish to the "settings.tasks.rules" property. The error that actually got logged in the console later is: 

![image](https://user-images.githubusercontent.com/35681649/230384976-dd28a474-fec7-4aef-9954-4905c1a55455.png)

I think it would be super useful to somehow include the error in the feedback doc as well.

Reply:
Thinking about this more - that log line contains the full compiled config. What if it is like 1MB? What if the log line happens many times (like each time the user navigates to tasks/targets tab). 

I agree it should be in the feedback doc but I'm also think we might want to truncate log lines

Reply:
We probably want to truncate all log lines to some decent value.

Reply:
Looking at prod data there are many arguments with >20000 chars today. I'm thinking to truncate at 1000 characters. It's kinda arbitrary.

Reply:
I think you can get a fairly good idea of what the error is in 1000 chars.